internal/xds/translator/ratelimit.go

```diff
@@ -159,30 +211,55 @@ func patchRouteWithRateLimitOnTypedFilterConfig(route *routev3.Route, rateLimits
 	if err != nil {
 		return fmt.Errorf("failed to marshal per-route ratelimit filter config: %w", err)
 	}
-	filterCfg[egv1a1.EnvoyFilterRateLimit.String()] = g
+	filterCfg[filterName] = g
 	return nil
 }
 
-func buildRouteRateLimits(descriptorPrefix string, global *ir.GlobalRateLimit) (rateLimits []*routev3.RateLimit) {
-	// Route descriptor for each route rule action
+// buildRouteRateLimits constructs rate limit actions for a given route based on the global rate limit configuration.
+func buildRouteRateLimits(route *ir.HTTPRoute) (rateLimits []*routev3.RateLimit) {
+	descriptorPrefix := route.Name
+	// Ensure route has rate limit config
+	if !routeContainsGlobalRateLimit(route) {
+		return nil
+	}
+
+	// Determine if the rate limit is shared across multiple routes.
+	global := route.Traffic.RateLimit.Global
+	isShared := isSharedRateLimit(route)
+
+	// Set the descriptor key based on whether the rate limit is shared.
+	var descriptorKey, descriptorValue string
+	if isShared {
+		// For shared rate limits, use BTP name as key and namespace as value
+		descriptorKey = getTrafficPolicyName(route)
```

Comment:
the current code sets the key and value to be the same value
```
				DescriptorKey:   getRouteDescriptor(descriptorPrefix),
				DescriptorValue: getRouteDescriptor(descriptorPrefix),
```

so it may be better to do the same for the shared case and use `traffic.Name` directly, we can optimize and improve debuggability and metric labels later

Reply:
+1 this reduce confusion.